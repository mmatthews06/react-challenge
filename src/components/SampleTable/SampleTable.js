import React from 'react';
import PropTypes from 'prop-types';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';

export default function SampleTable({ samples }) {
  return (
    <Paper>
      <Table>
        <TableHead>
          <TableRow>
            <TableCell align="center">Sample #</TableCell>
            <TableCell align="center">Sample Type</TableCell>
            <TableCell align="center">ID</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {samples.map((sample) => (
            <TableRow key={sample.ID}>
              <TableCell align="right">{sample['Sample Number']}</TableCell>
              <TableCell align="center">{sample.Type}</TableCell>
              <TableCell align="center">{sample.ID}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </Paper>
  );
}

SampleTable.propTypes = {
  samples: PropTypes.arrayOf(PropTypes.shape({
    'Sample Number': PropTypes.number,
    Type: PropTypes.string,
    ID: PropTypes.string,
  })).isRequired,
};
