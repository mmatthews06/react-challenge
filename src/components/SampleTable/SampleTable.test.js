/* global describe, expect, it */
import React from 'react';
import { shallow } from 'enzyme';

import SampleTable from './SampleTable';

describe('SampleTable', () => {
  it('should render an empty message', () => {
    expect(shallow(<SampleTable samples={[]} />)).toMatchSnapshot();
  });

  it('should match a snapshot', () => {
    expect(shallow(
      <SampleTable
        samples={[
          {
            'Sample Number': 1,
            Type: 'client',
            ID: 'ABC1234',
          },
          {
            'Sample Number': 2,
            Type: 'job',
            ID: 'DEF4567',
          },
        ]}
      />,
    )).toMatchSnapshot();
  });
});
