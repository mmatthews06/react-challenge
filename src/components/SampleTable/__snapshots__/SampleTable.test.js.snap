// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleTable should match a snapshot 1`] = `
<WithStyles(ForwardRef(Paper))>
  <WithStyles(ForwardRef(Table))>
    <WithStyles(ForwardRef(TableHead))>
      <WithStyles(ForwardRef(TableRow))>
        <WithStyles(ForwardRef(TableCell))
          align="center"
        >
          Sample #
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="center"
        >
          Sample Type
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="center"
        >
          ID
        </WithStyles(ForwardRef(TableCell))>
      </WithStyles(ForwardRef(TableRow))>
    </WithStyles(ForwardRef(TableHead))>
    <WithStyles(ForwardRef(TableBody))>
      <WithStyles(ForwardRef(TableRow))
        key="ABC1234"
      >
        <WithStyles(ForwardRef(TableCell))
          align="right"
        >
          1
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
        >
          client
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
        >
          ABC1234
        </WithStyles(ForwardRef(TableCell))>
      </WithStyles(ForwardRef(TableRow))>
      <WithStyles(ForwardRef(TableRow))
        key="DEF4567"
      >
        <WithStyles(ForwardRef(TableCell))
          align="right"
        >
          2
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
        >
          job
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
        >
          DEF4567
        </WithStyles(ForwardRef(TableCell))>
      </WithStyles(ForwardRef(TableRow))>
    </WithStyles(ForwardRef(TableBody))>
  </WithStyles(ForwardRef(Table))>
</WithStyles(ForwardRef(Paper))>
`;

exports[`SampleTable should render an empty message 1`] = `
<WithStyles(ForwardRef(Paper))>
  <WithStyles(ForwardRef(Table))>
    <WithStyles(ForwardRef(TableHead))>
      <WithStyles(ForwardRef(TableRow))>
        <WithStyles(ForwardRef(TableCell))
          align="center"
        >
          Sample #
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="center"
        >
          Sample Type
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="center"
        >
          ID
        </WithStyles(ForwardRef(TableCell))>
      </WithStyles(ForwardRef(TableRow))>
    </WithStyles(ForwardRef(TableHead))>
    <WithStyles(ForwardRef(TableBody)) />
  </WithStyles(ForwardRef(Table))>
</WithStyles(ForwardRef(Paper))>
`;
